core/src/test/java/io/confluent/rest/TestCustomizedHttpResponseHeaders.java

```diff
@@ -0,0 +1,284 @@
+      app.stop();
+    }
+  }
+
+  @Test(expected = ConfigException.class)
+  public void testInvalidHeaderConfigAction()throws Exception {
+    Properties props = createInvalidHeaderConfigActionProperties();
+    TestApp app = new TestApp(props);
+    CloseableHttpResponse response = null;
+    try {
+      app.start();
+      response = makeGetRequest(app, "/custom/resource1");
+    } finally {
+      try {
+        if (response != null) {
+          response.close();
+        }
+      } catch (Exception e) {
+      }
+      app.stop();
+    }
+  }
+
+  @Path("/custom")
+  @Produces(MediaType.TEXT_PLAIN)
+  public static class RestResource {
+    @GET
+    @Path("/resource1")
+    public String get1() {
+      return "testing resource1";
+    }
+    @POST
+    @Path("/resource2")
+    public String get2() {
+      return "testing resource2";
+    }
+  }
+
+  private static class TestApp extends Application<TestRestConfig> {
+    static Properties props = null;
+
+    public TestApp() {
+      super(createConfig());
+    }
+    public TestApp(Properties props) {
+      super(new TestRestConfig(props));
+      this.props = props;
+    }
+
+    @Override
+    public void setupResources(Configurable<?> config, TestRestConfig appConfig) {
+      config.register(new RestResource());
+    }
+
+    public String getUri() {
+      return (String)props.get(RestConfig.LISTENERS_CONFIG);
+    }
+
+    private static TestRestConfig createConfig() {
+      props = new Properties();
+      String uri = "http://localhost:8080";
+      return new TestRestConfig(props);
+    }
+  }
+
+  @SuppressWarnings("SameParameterValue")
+  private CloseableHttpResponse makeGetRequest(final TestApp app, final String path) throws Exception {
+    String uri = app.getUri();
+    final HttpGet httpget = new HttpGet(uri + path);
+    CloseableHttpClient httpClient = HttpClients.createDefault();
+    CloseableHttpResponse response = httpClient.execute(httpget);
+    try {
+      if (httpClient != null) {
+        httpClient.close();
+      }
+    } catch (Exception e) {
+    }
+    return response;
+  }
+
+  @SuppressWarnings("SameParameterValue")
+  private CloseableHttpResponse makePostRequest(final TestApp app, final String path) throws Exception {
+    String uri = app.getUri();
+    HttpPost httpPost = new HttpPost(uri + path);
+    CloseableHttpClient httpClient = HttpClients.createDefault();
+    CloseableHttpResponse response = httpClient.execute(httpPost);
+    try {
+      if (httpClient != null) {
+        httpClient.close();
+      }
+    } catch (Exception e) {
+    }
+    return response;
+  }
+
+  private String getResponseHeader(CloseableHttpResponse response, String name) {
+    String value = null;
+    Header[] header = response.getAllHeaders();
+    if (header != null && header.length > 0) {
+      for (int i = 0; i < header.length; i++) {
+        log.info("header name: {}, header value: {}.", header[i].getValue(), header[i].getValue());
+      }
+    }
```

Comment:
This is just for debugging, right? Should the log be at debug level? Also, can you use a simpler loop?
```
    for (Header h: response.getAllHeaders()) {
      log.info("header name: {}, header value: {}.", h.getValue(), h.getValue());
    }
```